app/controllers/coronavirus_form/have_you_been_made_unemployed_controller.rb

```diff
@@ -19,22 +19,24 @@ def submit
       flash.now[:validation] = invalid_fields
       log_validation_error(invalid_fields)
       render controller_path
-    elsif I18n.t("coronavirus_form.groups.being_unemployed.questions.have_you_been_made_unemployed.skip_next_question_options").include? @form_responses[:have_you_been_made_unemployed]
-      update_session_store
-      session[:questions_to_ask] = remove_questions(%w[are_you_off_work_ill])
-      redirect_path = next_question(controller_name)
-      redirect_path == "results" ? redirect_to_results : redirect_to(polymorphic_url(redirect_path))
-    elsif last_question == controller_name
-      redirect_to_results
     else
       update_session_store
-      session[:questions_to_ask] = add_questions(%w[are_you_off_work_ill], controller_name)
+      write_responses if last_question == controller_name
+      update_questions_to_ask
       redirect_to polymorphic_url(next_question(controller_name))
     end
   end
 
 private
 
+  def update_questions_to_ask
+    session[:questions_to_ask] = if I18n.t("coronavirus_form.groups.being_unemployed.questions.have_you_been_made_unemployed.skip_next_question_options").include? @form_responses[:have_you_been_made_unemployed]
+                                   remove_questions(%w[are_you_off_work_ill])
+                                 else
+                                   add_questions(%w[are_you_off_work_ill], controller_name)
+                                 end
+  end
+
   def redirect_to_results
```

Comment:
`redirect_to_results` is no longer being used and can be removed.

Reply:
Have taken this out